lib/ecto/schema.ex

```diff
@@ -1999,15 +1999,25 @@ defmodule Ecto.Schema do
   def __belongs_to__(mod, name, queryable, opts) do
     check_options!(opts, @valid_belongs_to_options, "belongs_to/3")
 
-    opts = Keyword.put_new(opts, :foreign_key, :"#{name}_id")
+    opts = Keyword.update(opts, :foreign_key, :"#{name}_id", &List.wrap/1)
```

Comment:
```suggestion
    opts = Keyword.update(opts, :foreign_key, [:"#{name}_id"], &List.wrap/1)
```

The function is not called if there is no value. The fact the above is wrong and there is no error makes me think the foreign key is always set? If so, use `Keyword.update!` instead and see what breaks (if anything at all).